protocols/v2/binary-sv2/codec/src/datatypes/non_copy_data_types/mod.rs

```diff
@@ -74,6 +74,228 @@ pub type B064K<'a> = Inner<'a, false, 1, 2, { u16::MAX as usize }>;
 /// represented using the `Inner` type with a 3-byte header.
 pub type B016M<'a> = Inner<'a, false, 1, 3, { 2_usize.pow(24) - 1 }>;
 
+impl fmt::Display for U32AsRef<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let inner = self.inner_as_ref();
+        write!(
+            f,
+            "U32AsRef({})",
+            u32::from_le_bytes([inner[0], inner[1], inner[2], inner[3]])
+        )
+    }
+}
+
+impl fmt::Display for B0255<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let inner = self
+            .inner_as_ref()
+            .iter()
+            .map(|byte| format!("{byte:02x}"))
+            .collect::<String>();
+        write!(f, "B0255({inner})")
+    }
+}
+
+impl fmt::Display for Sv2Option<'_, u32> {
+    // internally Sv2Option is pub struct Sv2Option<'a, T>(pub Vec<T>, PhantomData<&'a T>);
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let inner = self.to_owned().into_inner();
+        match inner {
+            Some(value) => write!(f, "Sv2Option({value})"),
+            None => write!(f, "Sv2Option(None)"),
+        }
+    }
+}
+
+impl Str0255<'_> {
+    /// Returns the value as a UTF-8 string if possible, otherwise as a hex string prefixed with 0x.
+    pub fn as_utf8_or_hex(&self) -> String {
+        match core::str::from_utf8(self.inner_as_ref()) {
+            Ok(s) => alloc::string::String::from(s),
+            Err(_) => format!(
+                "0x{}",
+                self.inner_as_ref()
+                    .iter()
+                    .map(|b| format!("{b:02x}"))
+                    .collect::<String>()
+            ),
+        }
+    }
+}
+
+impl fmt::Display for B064K<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let inner = self
+            .inner_as_ref()
+            .iter()
+            .map(|byte| format!("{byte:02x}"))
+            .collect::<String>();
+        write!(f, "B064K({inner})")
+    }
+}
+
+impl fmt::Display for U256<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let inner = self
+            .inner_as_ref()
+            .iter()
```

Comment:
```suggestion
            .iter()
            .rev()
```

this allows more intuitive display of `prev_hash` and `target` with 0s on the left side